File: LinkedList/linkedlist.py
class Node:
    def __init__(self, value):
        self.value = value
        self.next = None


class LinkedList:
    def __init__(self, value):
        self.head = Node(value)

    def traverse(self):
        current = self.head

        while current != None:
            print(current.value)
            current = current.next

    def append(self, value):
        current = self.head

        while current.next != None:
            current = current.next

        current.next = Node(value)

    def reverse(self):
        prev = None

        while self.head:
            next_node = self.head.next
            self.head.next = prev
            prev = self.head
            self.head = next_node

        self.head = prev


list = LinkedList(1)
list.append(2)
list.append(3)

print("LinkedList")
list.traverse()

print("Reversed LinkedList")
list.reverse()
list.traverse()

File: LinkedList/LinkedList.java
class Node<T> {
    public T value;
    public Node<T> next;

    public Node(T value) {
        this.value = value;
    }
}

public class LinkedList<T> {
    public Node<T> head;

    public LinkedList(T value) {
        this.head = new Node<T>(value);
    }

    public void traverse() {
        Node<T> current = head;
        while (current != null) {
            System.out.println(current.value);
            current = current.next;
        }
    }

    public void append(T value) {
        Node<T> current = head;
        while (current.next != null)
            current = current.next;
        current.next = new Node<T>(value);
    }

    public Node<T> pop() {
        Node<T> current = head;
        while (current.next.next != null)
            current = current.next;
        Node<T> popped = current.next;
        current.next = null;
        return popped;
    }

    public void reverse() {
        Node<T> prev = null;

        while (this.head != null) {
            Node<T> next_node = this.head.next;
            this.head.next = prev;
            prev = this.head;
            this.head = next_node;
        }

        this.head = prev;
    }

    public static void main(String[] args) {
        LinkedList<Integer> list = new LinkedList<Integer>(10);
        list.append(23);
        list.append(56);
        list.append(87);
        list.pop();

        System.out.println("LinkedList");
        list.traverse();
        list.reverse();

        System.out.println("Reversed LinkedList");
        list.traverse();
    }
}